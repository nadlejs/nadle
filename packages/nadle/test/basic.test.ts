import { it, describe } from "vitest";
import { exec, expectPass } from "setup";

describe("Basic", () => {
	it("can run a simple command", async () => {
		await expectPass(exec`hello`);
	});

	it("can run two commands sequentially", async () => {
		await expectPass(exec`hello goodbye`);
	});

	it("can run dependent task first", async () => {
		await expectPass(exec`copy`);
	});
});
