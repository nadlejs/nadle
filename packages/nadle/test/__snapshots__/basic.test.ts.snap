// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > single command > can run a simple command 3 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.basic.ts

[log] <Yellow>></Yellow> Task <Bold>node</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>node</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>install</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>install</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>compileSvg</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>compileSvg</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>compileTs</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>compileTs</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>compile</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>compile</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>5</BoldDim> tasks executed
`;

exports[`basic > single command > can run a simple command 4 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.basic.ts

[log] <Yellow>></Yellow> Task <Bold>node</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>node</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>install</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>install</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>test</BoldDim> <Yellow>STARTED</Yellow>

[log]
<Green>‚úì</Green> Task <Bold>test</BoldDim> <Green>DONE</Green> <Dim>{duration}</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>3</BoldDim> tasks executed
`;

exports[`basic > with --dry-run > should list tasks in order 1 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 2 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 3 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 4 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 5 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 6 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 7 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;

exports[`basic > with --dry-run > should list tasks in order 8 > stdout 1`] = `
[log] <Bold><Cyan>üõ†Ô∏è Welcome to Nadle {version}!</Cyan></BoldDim>
[log] Using Nadle from /ROOT/lib/index.js
[log] Loaded configuration from /ROOT/test/__fixtures__/main/nadle.abc.ts

[log] <Bold>Execution plan:</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-B</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.0</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.1</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C.2</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-A</BoldDim>
[log] <Yellow>></Yellow> Task <Bold>task-C</BoldDim>
[log]
<Bold><Green>RUN SUCCESSFUL</Green></BoldDim> in <Bold>{duration}</BoldDim>
[log] <Bold>0</BoldDim> task executed
`;
